futures-core/src/poll.rs

```diff
@@ -39,24 +42,23 @@ macro_rules! ready {
     })
 }
 
-/// Indicates whether a value is available, or if the current task has been
-/// scheduled for later wake-up instead.
-#[derive(Copy, Clone, Debug, PartialEq)]
-pub enum Poll<T> {
-    /// Represents that a value is immediately ready.
-    Ready(T),
 
-    /// Represents that a value is not ready yet.
-    ///
-    /// When a function returns `Pending`, the function *must* also
-    /// ensure that the current task is scheduled to be awoken when
-    /// progress can be made.
-    Pending,
+/// Temporary trait for inherent methods on `Poll` - should go away before merge
+pub trait PollExt<T> {
+    /// Change the ready value of this `Poll` with the closure provided
+    fn map<U, F>(self, f: F) -> Poll<U>
+        where F: FnOnce(T) -> U;
+    /// Returns whether this is `Poll::Ready`
+    fn is_ready(&self) -> bool;
+    /// Returns whether this is `Poll::Pending`
+    fn is_pending(&self) -> bool {
+        !self.is_ready()
```

Comment:
Your PR to core has been merged https://github.com/rust-lang/rust/pull/51442 These methods are now not required